pkg/objects/file.go

```diff
@@ -159,7 +165,13 @@ func (f File) PerformTransfer() TransferResult {
 	if body != nil {
 		defer body.Close()
 	}
-	if sourceState.SkipTransfer { // 304 Not Modified
+	if f.Job.Matcher.SimplisticComparison != nil && *f.Job.Matcher.SimplisticComparison {
+		targetSize := int64(hdr.SizeBytes().Get())
+		sourceSize := sourceState.SizeBytes
+		if sourceState.SkipTransfer && targetSize == sourceSize {
+			return TransferSkipped
+		}
```

Comment:
Upon re-reading this section, it occurred to me that this does not work.

1. If `sourceState.SkipTransfer` is true (which for both supported sources means a HTTP response with status code 304), this function can never return anything other than `TransferSkipped`. 304 responses never have a non-empty response body, so we couldn't use them to transfer the file even if we wanted.

2. Because 304 responses don't have a body, they either don't have a Content-Length header at all, or it will be zero, so `sourceSize` will always be 0 here and `targetSize == sourceSize` will never hold.

So first of all, we need a testcase for this. The testcase should run twice with simplistic comparison enabled, and look at the target `Last-Modified` to check that the files are not transferred twice.

Second, investigate how rclone is doing these checks, so that we can replicate them closely.